storage/innobase/fil/fil0crypt.cc

```diff
@@ -836,6 +836,65 @@ static inline void fil_crypt_read_crypt_data(fil_space_t *space) {
   mtr.commit();
 }
 
+/**
+Write crypt data belonging to space to page0
+@param space tablespace with crypt data to write
+*/
+static void fil_crypt_write_crypt_data_to_page0(fil_space_t *space) {
+  mtr_t mtr;
+  mtr.start();
+
+  if (buf_block_t *block = buf_page_get_gen(
+          page_id_t(space->id, 0), page_size_t(space->flags), RW_X_LATCH, NULL,
+          Page_fetch::NORMAL, __FILE__, __LINE__, &mtr)) {
+    space->crypt_data->write_page0(
```

Comment:
Because sometimes we want to first update page0 before we update the in-memory crypt_data struct. Please have a look at fil_crypt_flush_space and the place where it is called in fil_crypt_complete_rotate_space. We update crypt_data structure only when write to page0 was successful.